### ISSUE TITLE
docker::registry needs to change HOME env var based on local_user param

### ISSUE THREAD

**Issue by jstewart612** [puppetlabs/puppetlabs-docker#358]
## What you expected to happen?
When I changed local user param, the module would sudo to that user, use that user's home directory, and docker login.

## What happened?

It did all of this, except it hard-set HOME env var to /root which caused all execution to fail on obvious permissions errors because it tried to write files to the wrong home directory.

## How to reproduce it?

docker::registry as local_user anything but root.

## Anything else we need to know?
https://github.com/puppetlabs/puppetlabs-docker/blob/e833c292d67a3d3eae0b603c9c734b1db03d8a82/manifests/registry.pp#L58 is where the error is.  You need to write additional logic to detect actual user homedir and set that instead of /root.

## Versions:

```
$ puppet --version
5.5.6
$ docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:23:03 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:25:29 2018
  OS/Arch:          linux/amd64
  Experimental:     false
$ facter os
{
  architecture => "x86_64",
  family => "RedHat",
  hardware => "x86_64",
  name => "Scientific",
  release => {
    full => "7.4",
    major => "7",
    minor => "4"
  },
  selinux => {
    enabled => false
  }
}
$ puppet module list
/etc/puppetlabs/code/environments/production/modules (no modules installed)
/etc/puppetlabs/code/modules (no modules installed)
/opt/puppetlabs/puppet/modules (no modules installed)
```

## Logs:
```
Info: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/File[/root/registry-auth-puppet_receipt_docker-registry-useast2.tools.rentpath.com_deploy]: Scheduling refresh of Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: WARNING: Error loading config file: /root/.docker/config.json: stat /root/.docker/config.json: permission denied
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: WARNING! Using --password via the CLI is insecure. Use --password-stdin.
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: WARNING! Your password will be stored unencrypted in /root/.docker/config.json.
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: Configure a credential helper to remove this warning. See
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: https://docs.docker.com/engine/reference/commandline/login/#credentials-store
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns:
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: Error saving credentials: mkdir /root/.docker: permission denied
Notice: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]/returns: rm: cannot remove ‘/root/registry-auth-puppet_receipt_docker-registry-useast2.tools.rentpath.com_deploy’: Permission denied
Error: /Stage[main]/Rentpath::Services::Dockerce/Docker::Registry[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com]/Exec[deploy-harbor-infra@docker-registry-useast2.tools.rentpath.com auth]: Failed to call refresh: 'docker login -u 'harbor-infra' -p "${password}" docker-registry-useast2.tools.rentpath.com || rm -f "/root/registry-auth-puppet_receipt_docker-registry-useast2.tools.rentpath.com_deploy"' returned 1 instead of one of [0]
```

**Comment by shamil** [comment]
#353 should fix this, still not merged though

**Comment by davejrt** [comment]
Fixed in #353
